src/metabase/actions.clj

```diff
@@ -130,6 +132,21 @@
 
   nil)
 
+(defn database-for-action [action-or-id]
+  (toucan.models/do-post-select
```

Comment:
from @camsaul .

> There is no need to manually do-post-select or anything like that ever again, all that stuff is done with transducers now under the hood and is applied regardless of the query. You can even do (t2/select User ["SELECT * FROM user"]) and it will still do the post-select stuff on the results. Neat.

we will probably have a way to go around this without manually call `do-post-select` here

Reply:
Good point, I tried using toucan 2 for this. But if I'm using `t2/query-one`, it looks like I need to grab the current connection. It's a bit more work than I expected, so I wonder if @camsaul can advise on a more concise approach? https://github.com/metabase/metabase/pull/28153/commits/b9a3dcee7a0b65bda242a0c72b956e200325dc7c

Reply:
Pretty sure you want select-one here, not query-one. Either way if you pass nil as the connection to query-one it will use the default connection

Reply:
Updated to use `select-one`, thanks https://github.com/metabase/metabase/pull/28153/commits/4d586ab945c6d4bd392afcc143c94a66d91004f2